app/questionnaire/router.py

```diff
@@ -175,6 +186,13 @@ def get_previous_location_url(
 
         return None
 
+    def _get_return_to_location_url(self, location: Location, return_to: str) -> str:
+        if return_to == "section-summary":
+            return self._get_section_url(location)
+
+        if return_to == "final-summary" and self.is_questionnaire_complete:
+            return url_for("questionnaire.submit_questionnaire")
```

Comment:
Since previous URLs are rendered on the page and is clicked on by the users instead of us redirecting, the anchor link is no longer persisted when going back to the summary, so we will need to keep that in mind and think of a solution ad it will be needed for the upcoming return to summary journey.